packages/framework/src/Model/Product/Elasticsearch/ProductExportRepository.php

```diff
@@ -424,17 +425,19 @@ protected function getResults(QueryBuilder $queryBuilder, array $fields, int $do
      */
     protected function getVariantsForDefaultPricingGroup(Product $mainVariant, int $domainId): array
     {
-        if ($this->variantCache === null) {
-            $pricingGroup = $this->pricingGroupSettingFacade->getDefaultPricingGroupByDomainId($domainId);
-            $this->variantCache = $this->productRepository->getAllSellableVariantsByMainVariant($mainVariant, $domainId, $pricingGroup);
-        }
-
-        return $this->variantCache;
+        return $this->localCacheFacade->getOrSaveValue(
+            static::VARIANT_CACHE_NAMESPACE,
+            fn () => $this->productRepository->getAllSellableVariantsByMainVariant(
+                $mainVariant,
+                $domainId,
+                $this->pricingGroupSettingFacade->getDefaultPricingGroupByDomainId($domainId),
+            ),
+            static::VARIANT_CACHE_NAMESPACE,
```

Comment:
This key should be $mainVariantId - $pricingGroupId - $domainId right?

Reply:
no, in this case function getVariantsForDefaultPricingGroup should return **all** variants, so isn't  necessary make key as combination of this variables, we want just store and return all.